libs/librepcb/projecteditor/boardeditor/boardeditor.cpp

```diff
@@ -849,31 +848,52 @@ void BoardEditor::clearDrcMarker() noexcept {
   mGraphicsView->setSceneRectMarker(QRectF());
 }
 
-QStringList BoardEditor::getSearchToolBarCompleterList() noexcept {
-  QStringList list;
+QList<BI_Device*> BoardEditor::getSearchCandidates() noexcept {
+  QList<BI_Device*> candidates = {};
   if (Board* board = getActiveBoard()) {
-    foreach (const BI_Device* device, board->getDeviceInstances()) {
-      list.append(*device->getComponentInstance().getName());
+    foreach (BI_Device* device, board->getDeviceInstances()) {
+      candidates.append(device);
     }
```

Comment:
Could be simplified:

```suggestion
    candidates += board->getDeviceInstances().values();
```